tests/unit/s2n_client_hello_retry_test.c

```diff
@@ -958,7 +975,565 @@ int main(int argc, char **argv)
+         EXPECT_SUCCESS(s2n_client_hello_recv(server_conn));
+
+         EXPECT_SUCCESS(s2n_set_connection_hello_retry_flags(server_conn));
+
+         /* Custom s2n_server_hello_retry_send */
+         {
+             POSIX_CHECKED_MEMCPY(server_conn->handshake_params.server_random, hello_retry_req_random, S2N_TLS_RANDOM_DATA_LEN);
+
+             POSIX_GUARD(s2n_server_hello_write_message(server_conn));
+
+             /* Custom s2n_server_extensions_send */
+             {
+                 uint32_t data_available_before_extensions = s2n_stuffer_data_available(&server_conn->handshake.io);
+
+                 struct s2n_stuffer_reservation total_extensions_size = {0};
+                 POSIX_GUARD(s2n_stuffer_reserve_uint16(&server_conn->handshake.io, &total_extensions_size));
+
+                 /* Only send key share extension - exclude supported_versions */
+                 s2n_extension_send(&s2n_server_key_share_extension, server_conn, &server_conn->handshake.io);
+
+                 POSIX_GUARD(s2n_stuffer_write_vector_size(&total_extensions_size));
+
+                 if(s2n_stuffer_data_available(&server_conn->handshake.io) - data_available_before_extensions == sizeof(uint16_t)) {
+                     POSIX_GUARD(s2n_stuffer_wipe_n(&server_conn->handshake.io, sizeof(uint16_t)));
+                 }
+             }
+
+             /* Update transcript */
+             POSIX_GUARD(s2n_server_hello_retry_recreate_transcript(server_conn));
+
+             /* Reset handshake values */
+             server_conn->handshake.client_hello_received = 0;
+             server_conn->client_hello.parsed = 0;
+             POSIX_CHECKED_MEMSET((uint8_t*) server_conn->extension_requests_received, 0, sizeof(s2n_extension_bitfield));
+         }
+
+         EXPECT_SUCCESS(s2n_stuffer_wipe(&client_conn->handshake.io));
+         EXPECT_SUCCESS(s2n_stuffer_copy(&server_conn->handshake.io, &client_conn->handshake.io,
+                                         s2n_stuffer_data_available(&server_conn->handshake.io)));
+         client_conn->handshake.message_number = HELLO_RETRY_MSG_NO;
+
+         /* Client receives HelloRetryRequest */
+         EXPECT_FAILURE_WITH_ERRNO(s2n_server_hello_recv(client_conn),
+                                   S2N_ERR_BAD_MESSAGE);
+     }
+
+     /**
+     *= https://tools.ietf.org/rfc/rfc8446#4.1.4
+     *= type=test
+     *# Servers MUST ensure that they negotiate the
+     *# same cipher suite when receiving a conformant updated ClientHello (if
+     *# the server selects the cipher suite as the first step in the
+     *# negotiation, then this will happen automatically).
+     **/
+     {
+         DEFER_CLEANUP(struct s2n_cert_chain_and_key * chain_and_key, s2n_cert_chain_and_key_ptr_free);
+         EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key, S2N_DEFAULT_ECDSA_TEST_CERT_CHAIN,
+                                                        S2N_DEFAULT_ECDSA_TEST_PRIVATE_KEY));
```

Comment:
Something seems off about the spacing here?

Reply:
Fixed: a8c5a80

Reply:
Spacing is still strange-- too far to the right for this style of lining things up. Try a standard line wrap.

Search for other places you call `s2n_test_cert_chain_and_key_new` and `s2n_stuffer_copy`. Those seem to be the methods that are wrapped oddly.